### PR TITLE
Allow empty path elements (MLDB-1635)

### PR DESCRIPTION
**Pull request by jeremybarnes**
Self explanatory... this allows for paths to contain empty elements, just like JSON (which allows empty keys).

### CODE REVIEW

File: sql/testing/path_test.cc [hunk 1]
```diff
@@ -138,13 +130,23 @@ BOOST_AUTO_TEST_CASE(test_coords_parsing)
     }
 
     {
+        Path coords1 = Path::parse("..");
+        BOOST_CHECK_EQUAL(coords1.size(), 3);
+        BOOST_CHECK_EQUAL(coords1.toUtf8String(), "..");
```
Comment (nicolaskruchten) [inline sql/testing/path_test.cc]: should this not match to `""."".""` ?

**Comment by steadtler** [comment]
+1

File: sql/testing/path_test.cc (hunk 1 above)
Comment (jeremybarnes) [inline sql/testing/path_test.cc]: The "" aren't actually necessary as it's not ambiguous.  Do you think we should put them in anyway?

Comment (nicolaskruchten) [inline sql/testing/path_test.cc]: I think that it would be preferable to allowing `..` and `x.` or `.x` etc yes

**Pull request merged**